plugins/woocommerce-admin/client/marketplace/components/category-selector/category-selector.tsx

```diff
@@ -97,27 +145,38 @@ export default function CategorySelector(): JSX.Element {
 						className="woocommerce-marketplace__category-item"
 						key={ category.slug }
 					>
-						<CategoryLink { ...category } />
+						<CategoryLink
+							{ ...category }
+							selected={ category.slug === selected?.slug }
+						/>
 					</li>
 				) ) }
 				<li className="woocommerce-marketplace__category-item">
 					<CategoryDropdown
 						label={ __( 'More', 'woocommerce' ) }
 						categories={ secondBatch }
-						buttonClassName="woocommerce-marketplace__category-item-button"
+						buttonClassName={ classNames(
+							'woocommerce-marketplace__category-item-button',
+							{
+								'woocommerce-marketplace__category-item-button--selected':
+									isSelectedInSecondBatch(),
+							}
+						) }
 						contentClassName="woocommerce-marketplace__category-item-content"
 						arrowIconSize={ 20 }
+						selected={ selected }
 					/>
 				</li>
 			</ul>
 
 			<div className="woocommerce-marketplace__category-selector--full-width">
 				<CategoryDropdown
-					label={ __( 'All Categories', 'woocommerce' ) }
+					label={ mobileCategoryDropdownLabel() }
 					categories={ firstBatch.concat( secondBatch ) }
 					buttonClassName="woocommerce-marketplace__category-dropdown-button"
 					className="woocommerce-marketplace__category-dropdown"
 					contentClassName="woocommerce-marketplace__category-dropdown-content"
+					selected={ selected }
```

Comment:
I kinda don't like how many props I'm passing down to `CategoryDropdown`. There is probably a better way.